### PR TITLE
fix(typescript): set relations fields as optional

### PR DESCRIPTION
**Pull request by stafyniaksacha**

### CODE REVIEW

**Comment by maltejur** [comment]
https://github.com/stafyniaksacha/directus-extension-generate-types/blob/b6792dec42bce75bc8897137db9c747f89201ac3/src/lib/generateTypes/ts.ts#L60-L66

This part would also need to be updated accordingly

**Comment by stafyniaksacha** [comment]
@maltejur should be ok

File: src/lib/generateTypes/ts.ts [hunk 1]
```diff
@@ -57,11 +57,11 @@ function getType(field: Field, useIntersectionTypes = false) {
       field.relation.collection ? pascalCase(field.relation.collection) : "any"
     }${field.relation.type === "many" ? "[]" : ""}`;
   }
-  if (field.schema?.is_nullable) {
+  if (field.relation || field.schema?.is_nullable) {
     if (field.relation && useIntersectionTypes) {
-      type = `(${type}) | null`;
+      type = `(${type}) | null | undefined`;
```
Comment (maltejur) [inline src/lib/generateTypes/ts.ts:62]:
```suggestion
      type = `(${type}) | null`;
```

File: src/lib/generateTypes/ts.ts [hunk 2]
```diff
@@ -57,11 +57,11 @@ function getType(field: Field, useIntersectionTypes = false) {
...
     } else {
-      type += ` | null`;
+      type += ` | null | undefined`;
```
Comment (maltejur) [inline src/lib/generateTypes/ts.ts:64]:
```suggestion
      type += ` | null`;
```

**Review by maltejur — COMMENTED** [review]
I think the "undefined" is already covered by the `?` after field name.

**Comment by maltejur** [comment]
Thanks

**Pull request merged**